test/features/email/domain/usecases/view_attachment_for_web_interactor_test.dart

```diff
@@ -0,0 +1,253 @@
+import 'dart:async';
+
+import 'package:core/presentation/state/failure.dart';
+import 'package:core/presentation/state/success.dart';
+import 'package:dartz/dartz.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:jmap_dart_client/jmap/account_id.dart';
+import 'package:jmap_dart_client/jmap/core/id.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+import 'package:model/download/download_task_id.dart';
+import 'package:model/email/attachment.dart';
+import 'package:tmail_ui_user/features/email/domain/state/download_attachment_for_web_state.dart';
+import 'package:tmail_ui_user/features/email/domain/state/view_attachment_for_web_state.dart';
+import 'package:tmail_ui_user/features/email/domain/usecases/download_attachment_for_web_interactor.dart';
+import 'package:tmail_ui_user/features/email/domain/usecases/view_attachment_for_web_interactor.dart';
+
+import 'view_attachment_for_web_interactor_test.mocks.dart';
+
+@GenerateNiceMocks([
+  MockSpec<DownloadAttachmentForWebInteractor>(),
+  MockSpec<Failure>(),
+  MockSpec<Success>(),
+])
+void main() {
+  final downloadAttachmentForWebInteractor =
+      MockDownloadAttachmentForWebInteractor();
+  final tInteractor =
+      ViewAttachmentForWebInteractor(downloadAttachmentForWebInteractor);
+
+  final tTaskId = DownloadTaskId('123');
+  final tAttachment = Attachment();
+  final tAccountId = AccountId(Id('id'));
+  final tException = TimeoutException('321');
+  const tBaseDownloadUrl = 'base_download_url';
+  final tReceiveController = StreamController<Either<Failure, Success>>();
+  final tFailure = MockFailure();
+  final tSuccess = MockSuccess();
+  final tBytes = Uint8List(12);
```

Comment:
Please pay attention to the naming. It needs to be more clear.

Reply:
Done